SDWebImage/UIImageView+WebCache.m

```diff
@@ -69,17 +69,37 @@ - (void)sd_setImageWithURL:(nullable NSURL *)url
 #pragma mark - Animation of multiple images
 
 - (void)sd_setAnimationImagesWithURLs:(nonnull NSArray<NSURL *> *)arrayOfURLs {
+    return [self sd_setAnimationImagesWithURLs:arrayOfURLs progress:nil completed:nil];
+}
+
+- (void)sd_setAnimationImagesWithURLs:(nonnull NSArray<NSURL *> *)arrayOfURLs progress:(nullable SDImageBatchProgressBlock)progressBlock completed:(nullable SDImageBatchCompletionBlock)completedBlock {
     [self sd_cancelCurrentAnimationImagesLoad];
     NSPointerArray *operationsArray = [self sd_animationOperationArray];
     
+    NSUInteger totalCount = arrayOfURLs.count;
+    if (totalCount == 0) {
+        if (completedBlock) {
```

Comment:
NO. When the url is invalid or anything. (See our code which can return a `SDWebImageErrorInvalidURL`) You should always callback immediatelly on the caller queue. Instead of any queue dispatch. It's a common behavior, so this one follows